core/storage/trie/serialization/polkadot_codec.cpp

```diff
@@ -61,21 +67,23 @@ namespace kagome::storage::trie {
     return out;
   }
 
-  outcome::result<common::Buffer> PolkadotCodec::encodeNode(
-      const Node &node) const {
+  outcome::result<common::Buffer> PolkadotCodec::encodeNodeAndStoreChildren(
+      const Node &node, StoreChildren &store_children) const {
     switch (static_cast<TrieNode::Type>(node.getType())) {
       case TrieNode::Type::Leaf:
         return encodeLeaf(dynamic_cast<const LeafNode &>(node));
 
       case TrieNode::Type::BranchEmptyValue:
       case TrieNode::Type::BranchWithValue:
-        return encodeBranch(dynamic_cast<const BranchNode &>(node));
+        return encodeBranch(dynamic_cast<const BranchNode &>(node),
```

Comment:
Why not static_cast (with assert)?

Reply:
Didn't change that old code in this PR, don't know how it works